hbase-server/src/main/java/org/apache/hadoop/hbase/master/procedure/RSProcedureDispatcher.java

```diff
@@ -300,13 +318,28 @@ private boolean scheduleForRetry(IOException e) {
       if (numberOfAttemptsSoFar == 0 && unableToConnectToServer(e)) {
         return false;
       }
+      ExecuteProceduresRequest executeProceduresRequest = request.build();
+
+      // Check if the num of attempts have crossed the retry limit, and if the error type can
+      // fail-fast.
+      if (numberOfAttemptsSoFar >= failFastRetryLimit - 1 && isErrorTypeFailFast(e)) {
+        LOG
+          .warn("Number of retries {} exceeded limit {} for the given error type. Scheduling server"
+            + " crash for {}", numberOfAttemptsSoFar + 1, failFastRetryLimit, serverName, e);
+        // Expiring the server will schedule SCP and also reject the regionserver report from the
+        // regionserver if regionserver is somehow able to send the regionserver report to master.
+        // The master rejects the report by throwing YouAreDeadException, which would eventually
+        // result in the regionserver abort.
+        // This will also remove "serverName" from the ServerManager's onlineServers map.
+        master.getServerManager().expireServer(serverName);
+      }
       // Always retry for other exception types if the region server is not dead yet.
       if (!master.getServerManager().isServerOnline(serverName)) {
         LOG.warn("Request to {} failed due to {}, try={} and the server is not online, give up",
           serverName, e.toString(), numberOfAttemptsSoFar);
         return false;
       }
-      if (e instanceof RegionServerAbortedException || e instanceof RegionServerStoppedException) {
+      if (e instanceof RegionServerStoppedException) {
```

Comment:
Why we remove the RegionServerAbortedException here?

Reply:
Because `RegionServerAbortedException extends RegionServerStoppedException`